pkg/providers/subnet/subnet.go

```diff
@@ -64,41 +65,59 @@ func NewDefaultProvider(ec2api ec2iface.EC2API, cache *cache.Cache) *DefaultProv
 }
 
 func (p *DefaultProvider) List(ctx context.Context, nodeClass *v1beta1.EC2NodeClass) ([]*ec2.Subnet, error) {
```

Comment:
How do we make sure that we don't hold subnets in our cache forever when there is a delete operation that has been performed on a NodeClass? Thinking about it a little more -- if you hold the subnets in the cache like this, you basically have to do tracking to see which NodeClasses own the entry and respond to the delete operations for the NodeClasses, which sounds like a lot of work and sounds like it could create complicated semantics.

I'm thinking your original thought about relying on the status outright right now may actually be fine -- and we can move over to readiness conditions shortly. If we don't have subnets resolve -- like you were saying, we can just return no instance types and let the scheduler know that we can't use this NodePool. You could even consider the EKSNodeClass instanceProfile, security group, and subnet resolution as gates to return back valid values from the instance types call.

Reply:
The only problem then becomes how you ensure that if you are consistently re-reconciling the NodeClass -- that you still don't call the API too frequently when there is a request for data. Given that there is back-off built-into the NodeClass controller out-of-the-box are we good to just have each controller re-reconcile its own object and then hit the API cache to prevent duplicate requests (same filters on AMIs, subnets, or security groups) from causing a higher number of API requests.